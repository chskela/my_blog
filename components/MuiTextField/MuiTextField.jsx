import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

import classes from "./class.module.css";

const MuiTextField = React.memo(
  ({ icon, type, value, className, placeholder, onBlur, onChange }) => (
    <div className={cn(classes.textField, className)}>
      <input
        type={type}
        value={value}
        onBlur={onBlur}
        onChange={onChange}
        placeholder={placeholder}
      />
      {icon && <div data-role={"icon"}>{icon}</div>}
    </div>
  )
);

MuiTextField.propTypes = {
  icon: PropTypes.node,
  type: PropTypes.string,
  value: PropTypes.string,
  className: PropTypes.string,
  placeholder: PropTypes.string,
  onBlur: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

MuiTextField.defaultProps = {
  icon: null,
  type: "search",
  value: "",
  className: "",
  placeholder: "Search",
};

MuiTextField.displayName = "MuiTextField";

export default MuiTextField;
