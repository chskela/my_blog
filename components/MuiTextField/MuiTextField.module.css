.textField {
  position: relative;
}

.textField > input {
  font-size: var(--size-5);
  color: var(--color-black);
  border: 2px solid var(--color-secondary);
  box-sizing: border-box;
  border-radius: 10px;
  line-height: 25px;
  font-family: "Libre Baskerville", serif;
  padding: var(--size-3) var(--size-8);
  font-weight: 400;
  text-decoration: none;
  width: 100%;
  display: block;
}

.textField > div[data-role="icon"] {
  position: absolute;
  right: var(--size-3);
  top: var(--size-3);
}
