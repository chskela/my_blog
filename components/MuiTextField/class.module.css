.textField {
  position: relative;
}

.textField > input {
  width: 100%;
  outline: none;
  display: block;
  font-weight: 400;
  line-height: 25px;
  border-radius: 10px;
  text-decoration: none;
  box-sizing: border-box;
  font-size: var(--size-5);
  color: var(--color-black);
  padding: var(--size-3) var(--size-8);
  font-family: "Libre Baskerville", serif;
  border: 2px solid var(--color-secondary);
}

.textField > div[data-role="icon"] {
  top: var(--size-3);
  position: absolute;
  right: var(--size-3);
}
