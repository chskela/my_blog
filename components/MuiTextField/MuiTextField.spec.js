import React from "react";
import { shallow } from "enzyme";
import MuiTextField from "./MuiTextField";

const setUp = (props) => shallow(<MuiTextField {...props} />);

describe("MuiTextField component", () => {
  it("should render MuiTextField component default props", () => {
    const component = setUp({
      onChange: () => {},
      onBlur: () => {},
    });
    expect(component).toMatchSnapshot();
  });
  it("should render MuiTextField component whis props", () => {
    const component = setUp({
      onChange: () => {},
      onBlur: () => {},
      type: "text",
      value: "test",
      placeholder: "test",
      icon: "test",
      className: "test",
    });
    expect(component).toMatchSnapshot();
  });

  it(" method onChange must be called and the value changed", () => {
    const testState = { value: "" };
    const component = setUp({
      onBlur: () => {},
      onChange: (e) => {
        testState.value = e.value;
      },
      value: testState.value,
    });
    component.find("input").simulate("change", { value: "test" });
    expect(testState.value).toBe("test");
  });
  it("should call onBlur method", () => {
    const mockCallBack = jest.fn();
    const component = setUp({
      onChange: () => {},
      onBlur: mockCallBack,
    });
    component.find("input").simulate("blur");
    expect(mockCallBack.mock.calls.length).toBe(1);
  });
});
