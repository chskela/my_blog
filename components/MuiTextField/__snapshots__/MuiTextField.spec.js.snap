// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiTextField component should render MuiTextField component default props 1`] = `
<div
  className="textField"
>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Search"
    type="search"
    value=""
  />
</div>
`;

exports[`MuiTextField component should render MuiTextField component whis props 1`] = `
<div
  className="textField test"
>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="test"
    type="text"
    value="test"
  />
  <div
    data-role="icon"
  >
    test
  </div>
</div>
`;
